### DISCUSSION TITLE
pass component props when using as prop in V3

### DISCUSSION THREAD

**Discussion by zgid123** [chakra-ui/chakra-ui#9237] [Q&A] [answered]
Hi,

I am using `as` prop to render custom component for V3. But it seems the custom component's props are ignored in v3 unlike v2.

```ts
    <>
      <Box
        ref={ref}
        theme={theme}
        as={CodeMirror}
        extensions={[EditorView.lineWrapping, ...extensions]}
        basicSetup={{
          foldGutter: false,
          lineNumbers: false,
        }}
      />
      <CodeMirror
        extensions={[EditorView.lineWrapping, ...extensions]}
        basicSetup={{
          foldGutter: false,
          lineNumbers: false,
        }}
      />
    </>
```

The second CodeMirror will receive those props, but the first one didn't. This code is working for v2, but I still not find out the doc that mentioned this one and how to make it work for v3.

**Comment by segunadebayo** [comment, marked as answer]
`as={Component}` no longer forwards unknown props. Use `asChild` and put props on the child.
